client/src/components/Survey.js

```diff
@@ -2,82 +2,88 @@ import React, {useEffect, useState} from "react";
 import {Button, Form, Select, RangeSlider, FormLayout, TextField} from '@shopify/polaris';
 
 export function Survey({ survey = {} }) {
-    const [answers, setAnswers] = useState({});
+    const [questions, setQuestions] = useState([]);
 
     useEffect(() => {
-        if(survey.questions) {
-            survey.questions.forEach((question) => {
-                answers[question.question] = null
-                setAnswers(answers)
-                console.log('useeffect')
-            })
-        }
-        // eslint-disable-next-line
+      let qs = survey.questions || []
+      setQuestions(qs.map(q => {
+        q.answers = []
+        return q
+      }))
     }, [survey])
 
-    const questionMarkup = (question) => {
-        if(question.min){
-            // Range question
+    async function submitAnswers() {
+      const res = await fetch(`/api/answers`, {
+        method: "POST",
+        headers: {
+          "Accept": "application/json",
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify(questions)
+      });
+
+      await res
+        .json()
+        .catch(err => console.log(err));
+    }
+
+    const questionMarkup = (question, index) => {
+        switch(question.type) {
+          case "range":
             return (
                 <RangeSlider
-                    value={answers[question.question] == null ? parseInt(question.min) : parseInt(answers[question.question])}
+                    value={question.answers[0]?.val || parseInt(question.min)}
                     label={question.question}
                     onChange={(value, id) => {
-                        const tmp = {}
-                        tmp[question.question] = value
-                        setAnswers({...answers, ...tmp})
+                      questions[index].answers = [{val: value}]
+                      setQuestions([...questions])
```

Comment:
did you intend on pushing the changes you made when Hugh paired with you? I like the pattern of having a named handler. But it is not necessary to work

Reply:
I don't think it'd be worth it because you'd also have to pass in the question type to determine if it should be set as a value or response